server/.gitignore

```diff
@@ -1,2 +1,4 @@
 node_modules/
 npm-debug.log
+uploads/*
+!uploads/.gitkeep 
```

Comment:
Multer will create the destination folder for you, but you have to provide the option as a string (not a function):
> destination is used to determine within which folder the uploaded files should be stored. This can also be given as a string (e.g. '/tmp/uploads'). If no destination is given, the operating system's default directory for temporary files is used.

So your code would be like this instead:
```js
var storage = multer.diskStorage({

	destination: dir,

	filename: function (req, file, cb) {
		cb(null, file.fieldname + '-' + Date.now())
	}
})
```